src/main/java/com/minecolonies/colony/buildings/BuildingGuardTower.java

```diff
@@ -0,0 +1,187 @@
+
+    /**
+     * Constructor for the guardTower building.
+     *
+     * @param c Colony the building is in.
+     * @param l Location of the building.
+     */
+    public BuildingGuardTower(Colony c, BlockPos l)
+    {
+        super(c, l);
+    }
+
+    /**
+     * Gets the name of the schematic.
+     *
+     * @return Baker schematic name.
+     */
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return GUARD_TOWER;
+    }
+
+    /**
+     * Gets the max level of the baker's hut.
+     *
+     * @return The max level of the baker's hut.
+     */
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return GUARD_HUT_MAX_LEVEL;
+    }
+
+    /**
+     * The name of the baker's job.
+     *
+     * @return The name of the baker's job.
+     */
+    @NotNull
+    @Override
+    public String getJobName()
+    {
+        return GUARD_TOWER;
+    }
+
+    /**
+     * If no vision multiplier give health bonus.
+     * @return the bonus health.
+     */
+    public int getBonusHealth()
+    {
+        if (getBuildingLevel() > MAX_VISION_BONUS_MULTIPLIER)
+        {
+            return (getBuildingLevel() - MAX_VISION_BONUS_MULTIPLIER) * HEALTH_MULTIPLIER;
+        }
+        return 0;
+    }
+
+    @Override
+    public void onUpgradeComplete(final int newLevel)
+    {
+        if (this.getWorkerEntity() != null && newLevel > MAX_VISION_BONUS_MULTIPLIER)
+        {
+            this.getWorkerEntity().getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(BASE_MAX_HEALTH + getBonusHealth());
+        }
+
+        super.onUpgradeComplete(newLevel);
+    }
+
+    /**
+     * Getter for the bonus vision.
+     * @return an integer for the additional range.
+     */
+    public int getBonusVision()
+    {
+        if(getBuildingLevel() <= MAX_VISION_BONUS_MULTIPLIER)
+        {
+            return getBuildingLevel() * VISION_BONUS;
+        }
+        return MAX_VISION_BONUS_MULTIPLIER * VISION_BONUS;
+    }
+
+    @Override
+    public void setWorker(final CitizenData citizen)
+    {
+        if(citizen == null && this.getWorkerEntity() != null)
+        {
+            this.getWorkerEntity().getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(BASE_MAX_HEALTH);
+        }
+        else if(citizen != null && citizen.getCitizenEntity() != null)
+        {
+            citizen.getCitizenEntity().getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(BASE_MAX_HEALTH + getBonusHealth());
+        }
+        super.setWorker(citizen);
+    }
+
+    /**
+     * Create a Guard job.
+     *
+     * @param citizen the citizen to take the job.
+     * @return The new Guard job.
+     */
+    @NotNull
+    @Override
+    public AbstractJob createJob(CitizenData citizen)
+    {
+        return new JobGuard(citizen);
+    }
+
+    /**
+     * The client view for the baker building.
+     */
+    public static class View extends AbstractBuildingWorker.View
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2176)